### PR TITLE
feat: support community that all members allowed to pin

### LINKED ISSUE #15112: Support community with `:pinMessageAllMembersEnabled true`
related https://github.com/status-im/status-mobile/pull/14977#issuecomment-1433584320

### PR DESCRIPTION
**Pull request by yqrashawn**
fix https://github.com/status-im/status-mobile/issues/15112

support communities which allow members to pin messages

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/communities/core.cljs [hunk 1]
```diff
@@ -71,7 +71,10 @@
                         :canDeleteMessageForEveryone :can-delete-message-for-everyone?
                         :canJoin                     :can-join?
                         :requestedToJoinAt           :requested-to-join-at
-                        :isMember                    :is-member?})
+                        :isMember                    :is-member?
+                        :adminSettings               :admin-settings})
+      (update :admin-settings
+              #(set/rename-keys % {:pinMessageAllMembersEnabled :pin-messge-all-members-enabled?}))
```
Comment (J-Son89) [inline src/status_im/communities/core.cljs]: you have a typo in the key :pin-messge-all-members-enabled? -> :pin-message-all-members-enabled?

Comment (yqrashawn) [reply src/status_im/communities/core.cljs]: fixed

**Comment by VolodLytvynenko** [comment]
hi @yqrashawn Looks like some integration tests are failed https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-15197/5/consoleText Could you fix it, please?

File: src/status_im/communities/core.cljs [hunk 2]
```diff
@@ -71,7 +71,10 @@
                         :canDeleteMessageForEveryone :can-delete-message-for-everyone?
                         :canJoin                     :can-join?
                         :requestedToJoinAt           :requested-to-join-at
-                        :isMember                    :is-member?})
+                        :isMember                    :is-member?
+                        :adminSettings               :admin-settings})
+      (update :admin-settings
```
Comment (ilmotta) [inline src/status_im/communities/core.cljs:76]: @yqrashawn, as you probably know, here you can remove the wrapping function:

```clojure
(update :admin-settings set/rename-keys {:pinMessageAllMembersEnabled :pin-message-all-members-enabled?})
```

File: src/status_im2/contexts/chat/messages/list/view.cljs [hunk 3]
```diff
@@ -72,16 +72,20 @@
...
+        {:keys [pin-message-all-members-enabled?]} admin-settings
+        group-admin?                              (get admins current-public-key)
+        community-admin?                          (when community (community :admin))
+        message-pin-enabled                       (and (not public?)
+                                                       (or (not group-chat)
+                                                           (and group-chat
+                                                                (or group-admin?
+                                                                    pin-message-all-members-enabled?
+                                                                    community-admin?))))]
     {:group-chat                       group-chat
```
Comment (ilmotta) [inline src/status_im2/contexts/chat/messages/list/view.cljs:89]: I have the impression that this entire `get-render-data` function smells and it should be a subscription and properly unit tested. It's heavy on logic, doesn't render anything and it uses a bunch of other subscriptions, which is a clear use case for layer-3 subs.

**Review by ilmotta — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/list/view.cljs (hunk 3 above)
Comment (yqrashawn) [reply src/status_im2/contexts/chat/messages/list/view.cljs:89]: Agree, after a bit dig, this is much larger scope than I expected, affects a lot in the chat view.
I'll do some research and see what can be done. And, of course, in a separate issue.

Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/list/view.cljs:89]: You're right, too much for this PR. Thanks for digging into this problem :100:

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn thank you for PR. Here is the found issue

**Steps to reproduce:**
1. Go to the desktop (I used the last desktop master which has the ability to create communities where "any member can pin the message" option is available https://ci.infra.status.im/job/status-desktop/job/master/1528/)
2.  Go to 'create new community' flow -> set 'any member can pin the message' as enabled -> create community
image
3. Invite mobile user
4. Join to the community as a MOBILE user
5. Send a message
6. Long tap on the message

**Actual result:**
The 'pin to the channel' option is not shown
image

**Expected result:**
"Pin to the channel" option is shown 
image

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn could you rebase the PR, please?

**Comment by yqrashawn** [comment]
@VladimrLitvinenko rebased
Can you try on existing community? There's a similar issue when I implement this, which is, the allow-all-pin setting won't sync to the mobile app.  

I use two different mnemonics for dev. There are two kinds of communities.
One is communities that will appear on first time login. The others don't appear.
I changed the allow-all-pin setting of a first kind community on desktop.
Restarted the app on two simulators to make the setting sync. Can you try this?

**Comment by VolodLytvynenko** [comment]
> @VladimrLitvinenko rebased Can you try on existing community? There's a similar issue when I implement this, which is, the allow-all-pin setting won't sync to the mobile app.
> 
> I use two different mnemonics for dev. There are two kinds of communities. One is communities that will appear on first time login. The others don't appear. I changed the allow-all-pin setting of a first kind community on desktop. Restarted the app on two simulators to make the setting sync. Can you try this?

Hi @yqrashawn . I didn't understand about 2 kinds of communities :) Retested  with the following settings:
- Closed/Opened
- Encrypted/Non-encrypted
- All members can pin/Only admins can pin

No issues from my side. Issue 1 is fixed and PR is ready to be merged

**Pull request merged**